### DISCUSSION TITLE
How to install Selenium

### DISCUSSION THREAD

**Discussion by TheMissingPort** [UltimaHoarder/UltimaScraper#1002] [General]
Hello all

I apologize, I am very new to coding.

I had successfully been using V6.4.1, but now it seems like something is broken and I only get auth errors now, even after fixing the config settings.

Anyways, I downloaded V6.4.4, however I get an "ModuleNotFoundError: No module named 'seleniumwire'"

Now I've tried installing selenium with no luck. All I get is:
ERROR: Could not find a version that satisfies the requirement seleniumware
ERROR: No matching distribution found for seleniumware

I've looked up tutorials on YouTube, which are a bit lengthy but I tried. Is there something I'm missing? Does everyone go through these lengths to install Selenium? Or is it just me because I'm using a Silicon based Mac?

Any tips or suggestions would be greatly appreciated. Thank you

**Comment by lEtranger1942** [comment]
Have you tried pip install seleniumwire ? Or pip3 install seleniumwire ?

**Reply by TheMissingPort** [reply]
Yes that’s when I get the errors

**Comment by computerguy420** [comment]
The package name is: `selenium-wire`, not `seleniumwire`.

**Reply by TheMissingPort** [reply]
I tried that, I got:

ERROR: Command errored out with exit status 1:
     command: /Library/Frameworks/Python.framework/Versions/3.9/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/s3/zy9zzc9s3kqff4qbkkkhmqg80000gn/T/pip-install-drcfrtbp/cffi_ae006317dbe3403098f515998415b2d0/setup.py'"'"'; __file__='"'"'/private/var/folders/s3/zy9zzc9s3kqff4qbkkkhmqg80000gn/T/pip-install-drcfrtbp/cffi_ae006317dbe3403098f515998415b2d0/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/s3/zy9zzc9s3kqff4qbkkkhmqg80000gn/T/pip-record-lz_k7sr3/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.9/include/python3.9/cffi
         cwd: /private/var/folders/s3/zy9zzc9s3kqff4qbkkkhmqg80000gn/T/pip-install-drcfrtbp/cffi_ae006317dbe3403098f515998415b2d0/
    Complete output (48 lines):
    xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
    xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
    
        No working compiler found, or bogus compiler options passed to
        the compiler from Python's standard "distutils" module.  See
        the error messages above.  Likely, the problem is not related
        to CFFI but generic to the setup.py of any Python package that
        tries to compile C code.  (Hints: on OS/X 10.8, for errors about
        -mno-fused-madd see http://stackoverflow.com/questions/22313407/
        Otherwise, see https://wiki.python.org/moin/CompLangPython or
        the IRC channel #python on irc.freenode.net.)
    
        Trying to continue anyway.  If you are trying to install CFFI from
        a build done in a different context, you can ignore this warning.
    
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-universal2-3.9
    creating build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/backend_ctypes.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/error.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/setuptools_ext.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/__init__.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/cffi_opcode.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/vengine_gen.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/pkgconfig.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/model.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/ffiplatform.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/api.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/vengine_cpy.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/commontypes.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/lock.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/recompiler.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/cparser.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/verifier.py -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/_cffi_include.h -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/parse_c_type.h -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/_embedding.h -> build/lib.macosx-10.9-universal2-3.9/cffi
    copying cffi/_cffi_errors.h -> build/lib.macosx-10.9-universal2-3.9/cffi
    running build_ext
    building '_cffi_backend' extension
    creating build/temp.macosx-10.9-universal2-3.9
    creating build/temp.macosx-10.9-universal2-3.9/c
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g -I/usr/include/ffi -I/usr/include/libffi -I/Library/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c c/_cffi_backend.c -o build/temp.macosx-10.9-universal2-3.9/c/_cffi_backend.o -iwithsysroot/usr/include/ffi
    xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
    error: command '/usr/bin/gcc' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Library/Frameworks/Python.framework/Versions/3.9/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/s3/zy9zzc9s3kqff4qbkkkhmqg80000gn/T/pip-install-drcfrtbp/cffi_ae006317dbe3403098f515998415b2d0/setup.py'"'"'; __file__='"'"'/private/var/folders/s3/zy9zzc9s3kqff4qbkkkhmqg80000gn/T/pip-install-drcfrtbp/cffi_ae006317dbe3403098f515998415b2d0/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/s3/zy9zzc9s3kqff4qbkkkhmqg80000gn/T/pip-record-lz_k7sr3/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.9/include/python3.9/cffi Check the logs for full command output.
